File: app/src/main/java/com/hqq/example/ui/exception/ThrowActivity.java
package com.hqq.example.ui.exception;


import android.view.View;

import com.hqq.core.ui.base.BaseActivity;
import com.hqq.core.utils.log.LogUtils;
import com.hqq.example.R;


/**
 * @Author : huangqiqiang
 * @Package : com.hqq.coreapp.ui.activity
 * @FileName :   ThrowActivity
 * @Date : 2019/5/22 0022
 * @Email :  qiqiang213@gmail.com
 * @Descrive :异常
 */
public class ThrowActivity extends BaseActivity {


    @Override
    public int getLayoutViewId() {
        return R.layout.activity_throw;
    }

    @Override
    public void initView() {
    findViewById(R.id.button11).setOnClickListener(this::onViewClicked);
    findViewById(R.id.button12).setOnClickListener(this::onViewClicked);
    }


    public void onViewClicked(View view) {
        switch (view.getId()) {
            case R.id.button11:
                LogUtils.e(new Exception("一个异常 "));

                break;
            case R.id.button12:
                LogUtils.e(new RuntimeException("一个运行异常 "));
                break;

            default:
        }


    }
}

File: app/src/main/java/com/easy/example/ui/crash/ThrowActivity.kt
package com.easy.example.ui.crash

import android.view.View
import com.easy.core.ui.base.BaseActivity
import com.easy.core.utils.log.LogUtils.dInfo
import com.easy.example.R

/**
 * @Author : huangqiqiang
 * @Package : com.easy.coreapp.ui.activity
 * @FileName :   ThrowActivity
 * @Date : 2019/5/22 0022
 * @Email :  qiqiang213@gmail.com
 * @Describe :异常
 */
class ThrowActivity : BaseActivity() {


    override fun getLayoutViewId(): Int {
        return  R.layout.activity_throw
    }

    override fun initView() {
        findViewById<View>(R.id.button11).setOnClickListener { view: View -> onViewClicked(view) }
        findViewById<View>(R.id.button12).setOnClickListener { view: View -> onViewClicked(view) }
    }

    fun onViewClicked(view: View) {
        when (view.id) {
            R.id.button11 -> dInfo(Exception("一个异常 "))
            R.id.button12 -> dInfo(RuntimeException("一个运行异常 "))
            else -> {
            }
        }
    }
}